ops/framework.py

```diff
@@ -32,6 +32,27 @@
 from ops import charm
 from ops.storage import NoSnapshotError, SQLiteStorage
 
+if typing.TYPE_CHECKING:
```

Comment:
Out of curiosity, why does this need to be conditional?

Reply:
It's always false at (regular) run time.  And type checking tools set it to true - this allows for e.g. expensive type-checking-related setup to be skipped in production.

Reply:
also it helps avoid issues with circular imports and all: you're not really importing that module, you're just 'mentioning it' for type-checking purposes